GNUnet failed to build

## Issue description
Not sure if this is known, but I'm seeing the following error when trying to do `nix-env -i gnunet`.
```
...
  CCLD     perf-regex
  CC       gnunet-regex-profiler.o
  CCLD     gnunet-regex-profiler
/nix/store/4bqj6wzfq81lhqysh224lqbsi533cyin-binutils-2.26-dev/bin/ld: warning: libz.so.1, needed by ../../src/testbed/.libs/libgnunettestbed.so, not found (try using -rpath or -rpath-link)
../../src/testbed/.libs/libgnunettestbed.so: undefined reference to `uncompress'
../../src/testbed/.libs/libgnunettestbed.so: undefined reference to `compressBound'
../../src/testbed/.libs/libgnunettestbed.so: undefined reference to `compress2'
collect2: error: ld returned 1 exit status
Makefile:1110: recipe for target 'gnunet-regex-profiler' failed
make[3]: *** [gnunet-regex-profiler] Error 1
make[3]: Leaving directory '/tmp/nix-build-gnunet-0.10.1.drv-0/gnunet-0.10.1/src/regex'
Makefile:505: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/tmp/nix-build-gnunet-0.10.1.drv-0/gnunet-0.10.1/src'
Makefile:562: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-gnunet-0.10.1.drv-0/gnunet-0.10.1'
Makefile:451: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/5vz54n35hk58s8rk2v8ba1sv70f3z77g-gnunet-0.10.1.drv’ failed with exit code 2
error: build of ‘/nix/store/5vz54n35hk58s8rk2v8ba1sv70f3z77g-gnunet-0.10.1.drv’ failed
```

## Technical details

* System: Fedora 23
* Nix version: 1.11.2
* Nixpkgs version: 16.09pre81657.9445bb2

--- a/pkgs/applications/networking/p2p/gnunet/default.nix
+++ b/pkgs/applications/networking/p2p/gnunet/default.nix
@@ -20,21 +20,13 @@ stdenv.mkDerivation rec {
     # Brute force: since nix-worker chroots don't provide
     # /etc/{resolv.conf,hosts}, replace all references to `localhost'
     # by their IPv4 equivalent.
-    for i in $(find . \( -name \*.c -or -name \*.conf \) \
-                    -exec grep -l '\<localhost\>' {} \;)
-    do
-      echo "$i: substituting \`127.0.0.1' to \`localhost'..."
-      sed -i "$i" -e's/\<localhost\>/127.0.0.1/g'
-    done
+    find . \( -name \*.c -or -name \*.conf \) | \
+      xargs sed -ie 's|\<localhost\>|127.0.0.1|g'
 
     # Make sure the tests don't rely on `/tmp', for the sake of chroot
     # builds.
-    for i in $(find . \( -iname \*test\*.c -or -name \*.conf \) \
-                    -exec grep -l /tmp {} \;)
-    do
-      echo "$i: replacing references to \`/tmp' by \`$TMPDIR'..."
-      substituteInPlace "$i" --replace "/tmp" "$TMPDIR"
-    done
+    find . \( -iname \*test\*.c -or -name \*.conf \) | \
+      xargs sed -ie "s|/tmp|$TMPDIR|g"
 
     # Ensure NSS installation works fine
     configureFlags="$configureFlags --with-nssdir=$out/lib"
@@ -57,7 +49,7 @@ stdenv.mkDerivation rec {
     '';
   */
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "GNU's decentralized anonymous and censorship-resistant P2P framework";
 
     longDescription = ''
@@ -77,9 +69,9 @@ stdenv.mkDerivation rec {
 
     homepage = http://gnunet.org/;
 
-    license = stdenv.lib.licenses.gpl2Plus;
+    license = licenses.gpl2Plus;
 
-    maintainers = with stdenv.lib.maintainers; [ viric vrthra ];
-    platforms = stdenv.lib.platforms.gnu;
+    maintainers = with maintainers; [ viric vrthra ];
+    platforms = platforms.gnu;
   };
 }

--- a/pkgs/applications/networking/p2p/gnunet/default.nix
+++ b/pkgs/applications/networking/p2p/gnunet/default.nix
@@ -39,6 +39,10 @@ stdenv.mkDerivation rec {
     # Ensure NSS installation works fine
     configureFlags="$configureFlags --with-nssdir=$out/lib"
     patchShebangs src/gns/nss/install-nss-plugin.sh
+
+    sed -ie 's|@LDFLAGS@|@LDFLAGS@ $(Z_LIBS)|g' \
+      src/regex/Makefile.in \
+      src/fs/Makefile.in
   '';
 
   doCheck = false;
@@ -75,7 +79,7 @@ stdenv.mkDerivation rec {
 
     license = stdenv.lib.licenses.gpl2Plus;
 
-    maintainers = with stdenv.lib.maintainers; [ viric ];
+    maintainers = with stdenv.lib.maintainers; [ viric vrthra ];
     platforms = stdenv.lib.platforms.gnu;
   };
 }

